activejob/lib/active_job/queue_adapters/test_adapter.rb

```diff
@@ -26,15 +26,11 @@ def performed_jobs
       end
 
       def enqueue(job) #:nodoc:
-        return if filtered?(job)
-
         job_data = job_to_hash(job)
         perform_or_enqueue(perform_enqueued_jobs, job, job_data)
```

Comment:
What do you think about keeping the filtering logic in these two methods, just tweaking the boolean argument, that way we don't add that responsibility to `perform_or_enqueue`?

Reply:
@etiennebarrie it will be slower a bit, as in case of non perform condition `filtered?` will be called anyway. I'd leave it as it is, but in other case I can change `perform_or_enqueue` to the following interface `perform, job, job_data, filtered`.

Reply:
I don't think it would be any slower to have something like `perform_or_enqueue(perform_enqueued_jobs && !filtered?(job), job, job_data)`. 
Similarly to your changes, it would only call `filtered?` if `perform_enqueued_jobs` is false, it just keep the logic here rather than inside `perform_or_enqueue`.

Reply:
@etiennebarrie :+1: I will change it to your solution.